clap_complete/tests/testsuite/engine.rs

```diff
@@ -1191,3 +1191,55 @@ fn complete(cmd: &mut Command, args: impl AsRef<str>, current_dir: Option<&Path>
         .collect::<Vec<_>>()
         .join("\n")
 }
+
+#[test]
+fn suggest_value_path_with_mapper() {
+    let testdir = snapbox::dir::DirRoot::mutable_temp().unwrap();
+    let testdir_path = testdir.path().unwrap();
+    fs::write(testdir_path.join("a_file"), "").unwrap();
+    fs::write(testdir_path.join("b_file"), "").unwrap();
+    fs::create_dir_all(testdir_path.join("c_dir")).unwrap();
+    fs::create_dir_all(testdir_path.join("d_dir")).unwrap();
+    
+    fs::write(testdir_path.join("c_dir").join("Cargo.toml"), "").unwrap();
+    
+    let mut cmd = Command::new("dynamic")
+        .arg(
+            clap::Arg::new("input")
+                .long("input")
+                .short('i')
+                .add(ArgValueCompleter::new(
+                    PathCompleter::dir()
+                        .map(|candidate| {
+                            let path = Path::new(candidate.get_value());
+                            let cargo_toml_path = path.join("Cargo.toml");
+                            let has_cargo_toml = cargo_toml_path.exists();
+                            
+                            if has_cargo_toml {
+                                candidate
+                                    .help(Some("Addable cargo package".into()))
+                                    .display_order(Some(0))
+                            } else {
+                                candidate.display_order(Some(1))
+                            }
```

Comment:
Please demonstrate this feature when there is a user-provided directory `filter` *and* a `map` is used.

Reply:
And/or `PathCompleter::file` is being used with `map`